sysrib/server.go

```diff
@@ -375,106 +372,74 @@ func prefixString(prefix *sysribpb.Prefix) (string, error) {
-	}
-
-	buf := gopacket.NewSerializeBuffer()
-	if err := gopacket.SerializeLayers(buf, gopacket.SerializeOptions{}, ip, udp); err != nil {
-		return nil, fmt.Errorf("failed to serialize GUE headers: %v", err)
-	}
-
-	return []*fwdpb.ActionDesc{{
-		ActionType: fwdpb.ActionType_ACTION_TYPE_REPARSE,
-		Action: &fwdpb.ActionDesc_Reparse{
-			Reparse: &fwdpb.ReparseActionDesc{
-				HeaderId: headerID,
-				FieldIds: []*fwdpb.PacketFieldId{
-					{Field: &fwdpb.PacketField{FieldNum: fwdpb.PacketFieldNum_PACKET_FIELD_NUM_NEXT_HOP_IP}},
-					{Field: &fwdpb.PacketField{FieldNum: fwdpb.PacketFieldNum_PACKET_FIELD_NUM_PACKET_PORT_INPUT}},
-					{Field: &fwdpb.PacketField{FieldNum: fwdpb.PacketFieldNum_PACKET_FIELD_NUM_PACKET_PORT_OUTPUT}},
-				},
-				// After the UDP header, the rest of the packet (original packet) will be classified as payload.
-				Prepend: buf.Bytes(),
-			},
-		},
-	}}, nil
-}
-
 func resolvedRouteToRouteRequest(r *ResolvedRoute) (*dpb.Route, error) {
-	vrfID, err := niNameToVrfID(r.NIName)
-	if err != nil {
-		return nil, err
-	}
-
 	pfx, err := netip.ParsePrefix(r.Prefix)
 	if err != nil {
 		log.Errorf("Route prefix cannot be parsed: %v", err)
 		return nil, err
 	}
-
-	var nexthops []*dpb.NextHop
-	for nh := range r.Nexthops {
-		var actions []*fwdpb.ActionDesc
-		if nh.HasGUE() {
-			if actions, err = gueActions(pfx.Addr().Is4() || pfx.Addr().Is4In6(), nh.GUEHeaders); err != nil {
-				return nil, fmt.Errorf("error retrieving GUE actions: %v", err)
+	// Connected routes are routes with a single next hop with no address.
+	// TODO: Include a better signal for this.
+	if len(r.Nexthops) == 1 {
+		for nh := range r.Nexthops {
+			if nh.Address == "" {
+				return &dpb.Route{
+					Prefix: &dpb.RoutePrefix{
+						NetworkInstance: r.NIName,
+						Cidr:            r.Prefix,
+					},
+					Hop: &dpb.Route_Interface{
+						Interface: &dpb.OCInterface{
+							Interface:    nh.Port.Name,
+							Subinterface: nh.Port.Subinterface,
+						},
+					},
+				}, nil
 			}
 		}
+	}
+
+	nexthops := &dpb.NextHopList{}
+	for nh := range r.Nexthops {
 		dnh := &dpb.NextHop{
-			Dev: &dpb.NextHop_Port{
-				Port: nh.Port.Name,
+			Interface: &dpb.OCInterface{
+				Interface:    nh.Port.Name,
+				Subinterface: nh.Port.Subinterface,
 			},
-			Weight:             nh.Weight,
-			PreTransmitActions: actions,
+		}
+		if nh.HasGUE() {
+			if pfx.Addr().Is4() || pfx.Addr().Is4In6() {
+				dnh.Encap = &dpb.NextHop_Gue{
+					Gue: &dpb.GUE{
+						SrcIp:   nh.GUEHeaders.srcIP4[:],
+						DstIp:   nh.GUEHeaders.dstIP4[:],
+						DstPort: uint32(nh.GUEHeaders.dstPortv4),
+					},
+				}
+			} else {
+				dnh.Encap = &dpb.NextHop_Gue{
+					Gue: &dpb.GUE{
+						SrcIp:   nh.GUEHeaders.srcIP6[:],
+						DstIp:   nh.GUEHeaders.dstIP6[:],
+						DstPort: uint32(nh.GUEHeaders.dstPortv6),
+					},
+				}
+			}
 		}
 		if nh.Address != "" {
-			dnh.Ip = &dpb.NextHop_IpStr{IpStr: nh.Address}
+			dnh.NextHopIp = nh.Address
 		}
```

Comment:
`if` no longer needed? can include in declaration of `dnh`?

Reply:
done